firestore/google/cloud/firestore_v1/document.py

```diff
@@ -605,7 +605,8 @@ def id(self):
         Returns:
             str: The last component of the path of the document.
         """
-        return self._reference.id
+        if self._reference:
+            return self._reference.id
```

Comment:
The `else` case must also be handled.

Reply:
what have to return in else case?

Reply:
Could be None or exception, figure it out, please.